### PR TITLE
Questionnaire field type "Drop down" defaults to the blank option

### LINKED ISSUE #1457: Questionnaire field type "Drop Down" defaults to the first Option provided
When a user opens a questionnaire, the Drop Down fields are initially set to the first Option provided in the Questionnaire setup.

This is a problem because users may look past those fields and press submit, and their submission will reflect an answer they didn't give. It's also a problem because users aren't able to give NO answer on those questions, because there is no blank value.

The first option in a Drop Down field should be a blank option, that doesn't need to be set up in the Admin Panel. It appears in the Drop Down field when the user loads the page, and it is also a selectable option in the Drop Down field list.

If the user doesn't select anything, or selects the blank option, within the response appears in the data the same as on a Radio Button field where nothing is selected.

If the field is required, and the user doesn't select anything or selects the blank option, the user should be required to give a response that is NOT blank.

### LINKED ISSUE #1463: Blank responses/no response should not be included in results shown for polls
As uncovered in #1457 

image

Blank responses should not be included in results shown for polls. The above results should show [number of responses for the option]/[number of responses for all non-blank options].

### PR DESCRIPTION
**Pull request by Alihassanc5**
Closes #1457
Closes #1463 
 
- Questionnaire field type "Drop down" defaults to the blank option.
- Blank option is selectable.
- For required fields, user should be required to give a response that is NOT blank.
- Blank responses/no response excluded from results shown for polls.

image

- Results if blank option is selected

image

### CODE REVIEW

File: questionnaires/tests.py [hunk 1]
```diff
@@ -1023,3 +1024,59 @@ def test_csv_export_with_modified_questions(self):
...
+        self.site = SiteFactory(site_name='IoGT', port=8000, is_default_site=True)
+        self.home_page = self.site.root_page
+        self.poll = PollFactory(parent=self.home_page,)
+
+    def test_drop_down_defaults_to_the_blank_option(self):
+        PollFormFieldFactory(page=self.poll, field_type='dropdown', choices='c1|c2')
+
+        response = self.client.get(self.poll.url)
+        parsed_response = BeautifulSoup(response.content)
+        default_drop_down_option = parsed_response.find('div', {'class': 'quest-item'}).find('select').find('option').text
```
Comment (istride) [inline questionnaires/tests.py:1041]: I have a feeling that searching HTML markup like this is going to be harder to maintain as the number of tests increases. Beautiful Soup supports [CSS selectors](https://beautiful-soup-4.readthedocs.io/en/latest/#css-selectors), which may be more manageable.

The Selenium tests also use locators, so perhaps there is a way to share these locators across the test suite.

**Review by istride — CHANGES_REQUESTED** [review]
Please resolve conflicts.

File: questionnaires/tests.py [hunk 2]
```diff
@@ -1023,3 +1024,59 @@ def test_csv_export_with_modified_questions(self):
             f'{self.user_submission_02.id},Survey 01,2022-08-31 23:00:00+00:00,question_01,c2\r\n'
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(byte_response.decode(), expected_response)
+
+
+class PollTest(TestCase):
+    def setUp(self):
+        Site.objects.all().delete()
```
[the remark this answers is not in the document]
Comment (Alihassanc5) [reply questionnaires/tests.py]: Fixed it by simple approach.

File: questionnaires/tests.py [hunk 3]
```diff
@@ -1023,3 +1024,59 @@ def test_csv_export_with_modified_questions(self):
...
+        self.site = SiteFactory(site_name='IoGT', port=8000, is_default_site=True)
+        self.home_page = self.site.root_page
```
[the remark this answers is not in the document]
Comment (Alihassanc5) [reply questionnaires/tests.py]: Fixed it.

File: questionnaires/models.py [hunk 4]
```diff
@@ -624,42 +624,30 @@ def get_submission_class(self):
...
-                    # Just skip them.
-                    continue
-
-                question_stats = results.get(label, {})
-                if type(answer) != list:
-                    answer = [answer]
-
-                for answer_ in answer:
-                    question_stats[answer_] = question_stats.get(answer_, 0) + 1
+            for answer_ in answer:
```
[the remark this answers is not in the document]
Comment (Alihassanc5) [reply questionnaires/models.py]: Fixed it!

File: questionnaires/tests.py [hunk 5]
```diff
@@ -1023,3 +1024,59 @@ def test_csv_export_with_modified_questions(self):
...
+        self.assertEqual(len(results), 2)
+        self.assertEqual(results[0].select_one('.cust-check__title-left').text.strip(), 'c1')
+        self.assertEqual(" ".join(results[0].select_one('.cust-check__title-right').text.split()), '0 %')
+        self.assertEqual(results[1].select_one('.cust-check__title-left').text.strip(), 'c2')
+        self.assertEqual(" ".join(results[1].select_one('.cust-check__title-right').text.split()), '0 %')
+
+    def test_drop_down_non_blank_option_submission(self):
+        poll_question = PollFormFieldFactory(page=self.poll, field_type='dropdown', choices='c1|c2')
+
+        response = self.client.post(self.poll.url, {f'{poll_question.clean_name}': 'c1'})
```
Comment (istride) [inline questionnaires/tests.py:1073]: is this supposed to be an f-string?

**Review by istride — APPROVED** [review]
Please merge if the f-string issue is ok.

File: questionnaires/tests.py (hunk 5 above)
Comment (sheralim012) [reply questionnaires/tests.py]: No, updated it.

**Pull request merged**